tests/test_vqe.py

```diff
@@ -1027,7 +1033,11 @@ def circuit():
         res = qml.specs(circuit)()
 
         assert res["num_observables"] == 1
-        assert res["num_diagonalizing_gates"] == 0
+        assert (
+            res["num_diagonalizing_gates"] == 0
+            if not H.has_diagonalizing_gates
+            else len(H.diagonalizing_gates())
```

Comment:
Maybe this is overkill and we should just change it to 1 because that's what's expected in the new op_math version?

Reply:
yep I agree

Reply:
Is this the only point needing discussion / making a decision about `qml.specs`? I dont know in detail the context of where this function is used but sounds like a reasonable change to me now that we added `diagonalizing_gates` to `LinearCombination`.

I am going to copy-paste the test to highlight the difference to legacy H though 👍

Reply:
Plan from meeting: Will mark the test as xfail (leaving the expected result as "diagonalizing_gates: 0") for now, and make a shortcut story to revisit this

Reply:
sounds good! thanks! Feel free to merge afterwards